### DISCUSSION TITLE
Creating staticky linked program

### DISCUSSION THREAD

**Discussion by Detector-I** [fzyzcjy/flutter_rust_bridge#1074] [Q&A] [answered]
First of all thank you for the great library.
as you know flutter by default have a library name `flutter_windows.dll` (in windows) that will be loaded at runtime and get used, I wonder is it possible to generate a fully static project using rust <-> flutter?
I want my end program to just be a single executable...
I tried to search the internet, but couldn't find any good answer for my question...
(btw, having a discord (or zulip/mattermost) server would be very good, that way its way easier for everyone to be in touch and talk about this library...)

**Comment by fzyzcjy** [comment, marked as answer]
Hmm I am not sure even whether a pure-flutter program can be single-file. IIRC you still need to have a few vc runtime dll

**Reply by Detector-I** [reply]
yeah that's correct.
those VC runtimes are widely used in windows and other operation systems (games and other programs) so we can safely assume that user already have them.
my question is mostly about the flutter library itself, what I want to do is having a single program to ship to user (I want my program to be as portable as possible) so I want to try and link the flutter library directly to the rust program instead of having an a dll (or so...) beside the program.

**Reply by fzyzcjy** [reply]
Then that looks like a standard rust+flutter problem, so maybe search for that IMHO

**Reply by Detector-I** [reply]
> Then that looks like a standard rust+flutter problem, so maybe search for that IMHO

ok thank you.
sadly seems like flutter has been built with this type of interface in mind, I searched about statically linking it but so far couldn't find anything...
I asked this question here because I saw a program that used this library ([RustDest](https://github.com/rustdesk/rustdesk)) link the flutter statically (at least it seems like it...)
thanks anyway.

**Reply by fzyzcjy** [reply]
@Detector-I if so, then maybe ask on rustdesk? or maybe look at their open-source code
